lib/SPIRV/OCLUtil.cpp

```diff
@@ -107,6 +107,32 @@ size_t getAtomicBuiltinNumMemoryOrderArgs(StringRef Name) {
   return 1;
 }
 
+bool isComputeAtomicOCLBuiltin(StringRef DemangledName) {
+  if (!DemangledName.startswith(kOCLBuiltinName::AtomicPrefix) &&
+      !DemangledName.startswith(kOCLBuiltinName::AtomPrefix))
+    return false;
+
+  return llvm::StringSwitch<bool>(DemangledName)
+      .EndsWith("add", true)
+      .EndsWith("sub", true)
```

Comment:
It seems that we missed
`.EndsWith("cmpxchg", true)`

Reply:
xchg allows other then int result types. i.e. float. 
Also I think xchg doesn't do any computation, just access memory. So we don't need it here.